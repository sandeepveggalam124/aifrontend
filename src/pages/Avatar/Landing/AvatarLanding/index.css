.bg-last-avatarimage{
    background-size: cover;
    background-image: url('../../../../../../public/assests/images/Frame5.svg');
}
.avatar-landing-container{
    margin:0
    
}