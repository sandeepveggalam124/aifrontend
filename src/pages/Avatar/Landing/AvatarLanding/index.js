import React ,{useState,useRef}from 'react'
import {useLocation } from 'react-router-dom'
import ImageComponent from '../AvatarImage'
import WelcomeText from '../WelcomeText'
import GoLink from '../../../../components/home/GoLink'
import StartCapture from '../../Interview/StartCapture'
import StartInterview from '../../Interview/StartInterview'
import CameraComponent from '../../Interview/CaptureImage'
import CaptureIcons from '../../Interview/CaptureIcon'
import VideoComponent from '../../Interview/CaptureVideo'

const AvatarLanding = () => {
  const location=useLocation()
  const avatarSrc='/assests/images/avatar.svg'
  const avatarAlt='Avatar Image'
  const containerWidth='100%'
  const containerHeight='auto'

  const webcamRef=useRef(null);
  const [capturedImages,setCapturedImages]=useState([])
  const captureImage=()=>{
    if(webcamRef.current){
    const imageSrc=webcamRef.current.getScreenshot();
    setCapturedImages((prevImages)=>[...prevImages,imageSrc])
    console.log(capturedImages)
  }
  }

  const resetCapture=()=>{
    setCapturedImages([])
  }
  return (
    <div className="row custom-div text-white">
        <div className='col col-lg-1 border-end d-flex flex-column align-items-center'>
            <GoLink/>
        </div>
        <div className='col col-lg-5 border-end d-flex flex-column align-item-stretch p-0'>
            {location.pathname==='/avatarLanding' && <ImageComponent src={avatarSrc} alt={avatarAlt}/>}            
            {location.pathname==='/interview/startCapture' && <ImageComponent src={avatarSrc} alt={avatarAlt}/>}
            {location.pathname==='/interview/startInterview' && <ImageComponent src={avatarSrc} alt={avatarAlt}/>}                      
            {location.pathname==='/interview/captureIcons' && <CameraComponent  ref={webcamRef}/>}
            {location.pathname==='/interview/captureVideo' && <ImageComponent src={avatarSrc} alt={avatarAlt}/>}

        </div>
        <div className='col col-lg-5 bg-black d-flex flex-column justify-content-center align-items-center img-fluid'>
            {location.pathname==='/avatarLanding' && <WelcomeText/>}
            {location.pathname==='/interview/startCapture' && <StartCapture/>}
            {location.pathname==='/interview/startInterview' && <StartInterview/>}                      
            {location.pathname==='/interview/captureIcons' && <CaptureIcons onIconClick={captureImage} resetCapture={resetCapture}/>}
            {location.pathname==='/interview/captureVideo' && <VideoComponent ref={webcamRef} width={containerWidth} height={containerHeight} audio={false} showConfirmButton={true}/>}

        </div>
        <div className='col col-lg-1 bg-last-image'></div>
        
    </div>
  )
}

export default AvatarLanding