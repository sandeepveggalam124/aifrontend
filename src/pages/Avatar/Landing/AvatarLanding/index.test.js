import { MemoryRouter as Router } from "react-router-dom";
import {render} from "@testing-library/react";
import AvatarLanding from "./index";


it('renders correctly',()=>{
    const {container}=render(
        <Router>
            <AvatarLanding/>
        </Router>
    )
    expect(container).toMatchSnapshot()
})