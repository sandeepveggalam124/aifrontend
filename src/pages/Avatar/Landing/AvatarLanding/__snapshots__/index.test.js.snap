// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="row custom-div text-white"
  >
    <div
      class="col col-lg-1 border-end d-flex flex-column align-items-center"
    >
      <div
        class="row text-white"
      >
        <div
          class="col col-lg-12 d-flex align-items-center flex-column "
        >
          <div
            class="py-5"
          >
            <img
              alt="Bootstrap"
              src="/assests/images/profile.svg"
            />
          </div>
          <div
            class="rotate-text p-5"
          >
            <a
              class="welcome-text highlighted-link"
              href="link_to_gostandout"
            >
              GoStandOut
            </a>
          </div>
          <div
            class="rotate-text p-5"
          >
            <a
              class="welcome-text "
              href="link_to_golearn"
            >
              GoLearn
            </a>
          </div>
          <div
            class="rotate-text p-5"
          >
            <a
              class="welcome-text "
              href="link_to_gototest"
            >
              GoToTest
            </a>
          </div>
          <div
            class="rotate-text p-5"
          >
            <a
              class="welcome-text "
              href="link_to_gosolvecode"
            >
              GoSolveCode
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col col-lg-5 border-end d-flex flex-column align-item-stretch p-0"
    />
    <div
      class="col col-lg-5 bg-black d-flex flex-column justify-content-center align-items-center img-fluid"
    />
    <div
      class="col col-lg-1 bg-last-image"
    />
  </div>
</div>
`;
